src/main/host/descriptor/regular_file.c

```diff
@@ -60,7 +72,15 @@ struct _RegularFile {
             int flagsAtOpen;
             /* The permission mode the file was opened with. */
             mode_t modeAtOpen;
-            /* The path of the file when it was opened. */
+            /* Callback to (re)-generate contents. */
+            _generateInMemoryFileContentCbType generate_contents_cb;
+            /* Whether contents should be re-generated on an lseek operation.
+             * Typical for some? all? proc files)
+             *
+             * TODO: Actually implement lseek for inMemoryFile. Currently
+             * just returns an error.
+             */
+            bool regen_after_lseek;
```

Comment:
This variable is currently not read. Just wanted to flag in case you think it's better to add this bit alongside an lseek implementation instead of now.

I'm OK with keeping it in now too, because you've already worked it out and so we wouldn't have to remember how to track it later.

Reply:
Yeah I had originally planned to implement the regeneration, before realizing that lseek isn't implemented at all for these files. It would be terribly difficult, but doing it + adding the appropriate tests was a little more scope creep than I'd bargained for. But yeah, I figured might as well still leave in the bits that will make implementing the regeneration trivial once lseek is implemented.